<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Main_Window</class>
 <widget class="QMainWindow" name="Main_Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="Extended_TabWidget" name="tabWidget">
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>640</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menu_Open_Recent">
     <property name="title">
      <string>Open &amp;Recent</string>
     </property>
     <property name="icon">
      <iconset theme="document-open-recent">
       <normaloff/>
      </iconset>
     </property>
    </widget>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="menu_Open_Recent"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="action_Save_As"/>
    <addaction name="action_Save_All"/>
    <addaction name="action_Export"/>
    <addaction name="separator"/>
    <addaction name="action_Close"/>
    <addaction name="action_Close_All"/>
    <addaction name="separator"/>
    <addaction name="action_Print"/>
    <addaction name="action_Page_Setup"/>
    <addaction name="action_Print_Preview"/>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_Undo"/>
    <addaction name="action_Redo"/>
    <addaction name="separator"/>
    <addaction name="action_Cut"/>
    <addaction name="action_Copy"/>
    <addaction name="action_Paste"/>
    <addaction name="action_Delete"/>
    <addaction name="separator"/>
    <addaction name="action_Select_All"/>
    <addaction name="separator"/>
    <addaction name="action_Preferences"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <widget class="QMenu" name="menu_Toolbars">
     <property name="title">
      <string>&amp;Toolbars</string>
     </property>
     <property name="icon">
      <iconset theme="configure-toolbars">
       <normaloff/>
      </iconset>
     </property>
    </widget>
    <widget class="QMenu" name="menu_Docks">
     <property name="title">
      <string>&amp;Dialogs</string>
     </property>
     <property name="icon">
      <iconset theme="preferences-system-windows-move">
       <normaloff/>
      </iconset>
     </property>
    </widget>
    <widget class="QMenu" name="menu_Zoom">
     <property name="title">
      <string>&amp;Zoom</string>
     </property>
     <property name="icon">
      <iconset theme="page-zoom">
       <normaloff/>
      </iconset>
     </property>
     <addaction name="action_Zoom_In"/>
     <addaction name="action_Zoom_Out"/>
     <addaction name="action_Reset_Zoom"/>
    </widget>
    <addaction name="menu_Toolbars"/>
    <addaction name="menu_Docks"/>
    <addaction name="separator"/>
    <addaction name="menu_Zoom"/>
    <addaction name="action_Reset_View"/>
    <addaction name="separator"/>
    <addaction name="action_Show_Graph"/>
    <addaction name="action_Display_Knot"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menu_Nodes">
    <property name="title">
     <string>&amp;Nodes</string>
    </property>
    <addaction name="action_Connect"/>
    <addaction name="action_Disconnect"/>
    <addaction name="action_Merge"/>
    <addaction name="action_Erase"/>
    <addaction name="separator"/>
    <addaction name="action_Snap_to_Grid"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_Connected"/>
    <addaction name="separator"/>
    <addaction name="action_Mirror_Horizontal"/>
    <addaction name="action_Mirror_Vertical"/>
    <addaction name="separator"/>
    <addaction name="action_Rotate"/>
    <addaction name="action_Scale"/>
   </widget>
   <widget class="QMenu" name="menu_Tools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="action_Edit_Graph"/>
    <addaction name="action_Edge_Loop"/>
    <addaction name="separator"/>
    <addaction name="action_Insert_Polygon"/>
    <addaction name="separator"/>
    <addaction name="action_Refresh_Path"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_Manual"/>
    <addaction name="separator"/>
    <addaction name="action_Report_Bugs"/>
    <addaction name="separator"/>
    <addaction name="action_About_Qt"/>
    <addaction name="action_About"/>
   </widget>
   <widget class="QMenu" name="menu_Plugins">
    <property name="title">
     <string>&amp;Plugins</string>
    </property>
    <widget class="QMenu" name="menu_Insert">
     <property name="toolTip">
      <string>Insert objects from plugin</string>
     </property>
     <property name="title">
      <string>&amp;Insert</string>
     </property>
     <property name="icon">
      <iconset theme="preferences-plugin-script">
       <normaloff/>
      </iconset>
     </property>
    </widget>
    <widget class="QMenu" name="menu_Transform">
     <property name="toolTip">
      <string>Transform selection with plugin</string>
     </property>
     <property name="title">
      <string>&amp;Transform</string>
     </property>
     <property name="icon">
      <iconset theme="preferences-plugin-script">
       <normaloff/>
      </iconset>
     </property>
    </widget>
    <addaction name="menu_Insert"/>
    <addaction name="menu_Transform"/>
    <addaction name="separator"/>
    <addaction name="action_Configure_Plugins"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
   <addaction name="menu_View"/>
   <addaction name="menu_Nodes"/>
   <addaction name="menu_Tools"/>
   <addaction name="menu_Plugins"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolbar_main">
   <property name="windowTitle">
    <string>Main Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
   <addaction name="action_Save_As"/>
   <addaction name="action_Export"/>
   <addaction name="separator"/>
   <addaction name="action_Undo"/>
   <addaction name="action_Redo"/>
   <addaction name="separator"/>
   <addaction name="action_Cut"/>
   <addaction name="action_Copy"/>
   <addaction name="action_Paste"/>
  </widget>
  <widget class="QToolBar" name="toolbar_view">
   <property name="windowTitle">
    <string>View Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Zoom_In"/>
   <addaction name="action_Zoom_Out"/>
   <addaction name="separator"/>
   <addaction name="action_Show_Graph"/>
  </widget>
  <widget class="QToolBar" name="toolbar_draw">
   <property name="windowTitle">
    <string>Drawing Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Edit_Graph"/>
   <addaction name="action_Edge_Loop"/>
   <addaction name="separator"/>
   <addaction name="action_Connect"/>
   <addaction name="action_Disconnect"/>
   <addaction name="action_Merge"/>
   <addaction name="action_Erase"/>
   <addaction name="separator"/>
   <addaction name="action_Insert_Polygon"/>
  </widget>
  <widget class="QToolBar" name="toolbar_transform">
   <property name="windowTitle">
    <string>Transformation Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Mirror_Horizontal"/>
   <addaction name="action_Mirror_Vertical"/>
   <addaction name="separator"/>
   <addaction name="action_Rotate"/>
   <addaction name="action_Scale"/>
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Open...</string>
   </property>
  </action>
  <action name="action_New">
   <property name="icon">
    <iconset theme="document-new">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="action_Save_As">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Save &amp;As...</string>
   </property>
  </action>
  <action name="action_Save_All">
   <property name="icon">
    <iconset theme="document-save-all">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Save A&amp;ll</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_Print">
   <property name="icon">
    <iconset theme="document-print">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Print...</string>
   </property>
  </action>
  <action name="action_Page_Setup">
   <property name="icon">
    <iconset theme="document-properties">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Pa&amp;ge Setup...</string>
   </property>
  </action>
  <action name="action_Print_Preview">
   <property name="icon">
    <iconset theme="document-print-preview">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Print Pre&amp;view</string>
   </property>
  </action>
  <action name="action_Close">
   <property name="icon">
    <iconset theme="tab-close">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Close</string>
   </property>
  </action>
  <action name="action_Close_All">
   <property name="text">
    <string>Close All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+W</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="action_Copy">
   <property name="icon">
    <iconset theme="edit-copy">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Copy</string>
   </property>
  </action>
  <action name="action_Paste">
   <property name="icon">
    <iconset theme="edit-paste">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Paste</string>
   </property>
  </action>
  <action name="action_Cut">
   <property name="icon">
    <iconset theme="edit-cut">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Cu&amp;t</string>
   </property>
  </action>
  <action name="action_Delete">
   <property name="icon">
    <iconset theme="edit-delete">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Delete</string>
   </property>
  </action>
  <action name="action_Select_All">
   <property name="icon">
    <iconset theme="edit-select-all">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Select &amp;All</string>
   </property>
  </action>
  <action name="action_Preferences">
   <property name="icon">
    <iconset theme="preferences-other">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Preferences...</string>
   </property>
  </action>
  <action name="action_Reset_View">
   <property name="icon">
    <iconset theme="view-restore">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Reset View</string>
   </property>
  </action>
  <action name="action_Zoom_In">
   <property name="icon">
    <iconset theme="zoom-in">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;In</string>
   </property>
  </action>
  <action name="action_Zoom_Out">
   <property name="icon">
    <iconset theme="zoom-out">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;Out</string>
   </property>
  </action>
  <action name="action_Reset_Zoom">
   <property name="icon">
    <iconset theme="zoom-original">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Reset Zoom</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="action_Show_Graph">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="knot-graph-on">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Display &amp;Graph</string>
   </property>
  </action>
  <action name="action_Display_Knot">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Display &amp;Knot</string>
   </property>
  </action>
  <action name="action_Connect">
   <property name="icon">
    <iconset theme="format-connect-node">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Connect</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="action_Disconnect">
   <property name="icon">
    <iconset theme="format-disconnect-node">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Disconnect</string>
   </property>
  </action>
  <action name="action_Mirror_Horizontal">
   <property name="icon">
    <iconset theme="object-flip-horizontal">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Horizontal Flip</string>
   </property>
  </action>
  <action name="action_Mirror_Vertical">
   <property name="icon">
    <iconset theme="object-flip-vertical">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Vertical Flip</string>
   </property>
  </action>
  <action name="action_Rotate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="transform-rotate">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Rotate</string>
   </property>
  </action>
  <action name="action_Scale">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="transform-scale">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Scale</string>
   </property>
  </action>
  <action name="action_Merge">
   <property name="icon">
    <iconset theme="format-join-node">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Merge</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="action_Erase">
   <property name="icon">
    <iconset theme="format-remove-node">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Erase</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_Edit_Graph">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="draw-line">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Edit Graph</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+E</string>
   </property>
  </action>
  <action name="action_Edge_Loop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="draw-jagged-line">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Edge &amp;Loop</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+L</string>
   </property>
  </action>
  <action name="action_Insert_Polygon">
   <property name="icon">
    <iconset theme="draw-polygon">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Insert &amp;Polygon...</string>
   </property>
  </action>
  <action name="action_Refresh_Path">
   <property name="icon">
    <iconset theme="view-refresh">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Refresh Path</string>
   </property>
  </action>
  <action name="action_Manual">
   <property name="icon">
    <iconset theme="help-contents">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Manual</string>
   </property>
  </action>
  <action name="action_About">
   <property name="icon">
    <iconset theme="help-about">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;About Knotter...</string>
   </property>
  </action>
  <action name="action_Export">
   <property name="icon">
    <iconset theme="document-export">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>E&amp;xport...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="action_Snap_to_Grid">
   <property name="icon">
    <iconset theme="snap-orthogonal">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Snap to &amp;Grid</string>
   </property>
  </action>
  <action name="action_Undo">
   <property name="icon">
    <iconset theme="edit-undo">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="action_Redo">
   <property name="icon">
    <iconset theme="edit-redo">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
  </action>
  <action name="action_Configure_Plugins">
   <property name="icon">
    <iconset theme="preferences-plugin">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Configure Plugins...</string>
   </property>
  </action>
  <action name="action_About_Qt">
   <property name="text">
    <string>About &amp;Qt...</string>
   </property>
  </action>
  <action name="action_Report_Bugs">
   <property name="icon">
    <iconset theme="tools-report-bug">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Report &amp;Bugs..</string>
   </property>
  </action>
  <action name="actionSelect_Connected">
   <property name="text">
    <string>Select Connected</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Extended_TabWidget</class>
   <extends>QTabWidget</extends>
   <header>extended_tab_widget.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>tabWidget</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>Main_Window</receiver>
   <slot>close_tab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>-84</x>
     <y>160</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_New</sender>
   <signal>triggered()</signal>
   <receiver>Main_Window</receiver>
   <slot>create_tab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>close_tab(int)</slot>
  <slot>create_tab()</slot>
  <slot>change_rendering()</slot>
 </slots>
</ui>
